### PR TITLE
Problem: hermes is not supported on 'x86_64-darwin'

### LINKED ISSUE #198: Problem: hermes is not supported on 'x86_64-darwin'
**Describe the bug**

After pulling the commit 9c3ba89b259f6da54a2c9f6096c8f0dd496efdc4, then
run `nix-shell integration_tests/shell.nix`, it prints out the following error:
```
error: Package ‘hermes’ in /Users/damonchen/Documents/cronos/nix/hermes.nix:4 is not supported on ‘x86_64-darwin’, refusing to evaluate.

a) To temporarily allow packages that are unsupported for this system, you can use an environment variable
   for a single invocation of the nix tools.

     $ export NIXPKGS_ALLOW_UNSUPPORTED_SYSTEM=1

b) For `nixos-rebuild` you can set
  { nixpkgs.config.allowUnsupportedSystem = true; }
in configuration.nix to override this.

c) For `nix-env`, `nix-build`, `nix-shell` or any other Nix command you can add
  { allowUnsupportedSystem = true; }
to ~/.config/nixpkgs/config.nix.

(use '--show-trace' to show detailed location information)

```

As it states, run `export NIXPKGS_ALLOW_UNSUPPORTED_SYSTEM=1` can temporarily avoid it.

**Desktop (please complete the following information):**
 - OS: macOS Big Sur
 - Version 11.6.1

### PR DESCRIPTION
**Pull request by damoncro**
👮🏻👮🏻👮🏻 !!!! REFERENCE THE PROBLEM YOUR ARE SOLVING IN THE PR TITLE AND DESCRIBE YOUR SOLUTION HERE !!!! DO NOT FORGET !!!! 👮🏻👮🏻👮🏻

Close #198

1. Support 0.49 dapptools https://github.com/dapphub/dapptools/releases/tag/hevm%2F0.49.0 (Use `dapp` cachix: https://app.cachix.org/cache/dapp)
2. Support 0.8 herems  https://github.com/informalsystems/ibc-rs/releases/tag/v0.8.0 (Both mac and linux)
3. Improve test-ibc. 
    - Enable REST and use wait_for_port rather than time.sleep.
    - Fix `get_balance` which did not specify data path 
    - Fix a warning: Replace `defaultAccount` with `default_account` 

Test Result:
https://github.com/chenyanming/cronos/runs/4282262307?check_suite_focus=true
Mac: 
image

# PR Checklist:

- [x] Have you read the [CONTRIBUTING.md](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md)?
- [x] Does your PR follow the [C4 patch requirements](https://rfc.zeromq.org/spec:42/C4/#23-patch-requirements)?
- [x] Have you rebased your work on top of the latest master? 
- [x] Have you checked your code compiles? (`make`)
- [ ] Have you included tests for any non-trivial functionality?
- [x] Have you checked your code passes the unit tests? (`make test`)
- [ ] Have you checked your code formatting is correct? (`go fmt`)
- [ ] Have you checked your basic code style is fine? (`golangci-lint run`)
- [ ] If you added any dependencies, have you checked they do not contain any known vulnerabilities? (`go list -json -m all | nancy sleuth`)
- [ ] If your changes affect the client infrastructure, have you run the integration test?
- [ ] If your changes affect public APIs, does your PR follow the [C4 evolution of public contracts](https://rfc.zeromq.org/spec:42/C4/#26-evolution-of-public-contracts)?
- [ ] If your code changes public APIs, have you incremented the crate version numbers and documented your changes in the [CHANGELOG.md](https://github.com/crypto-org-chain/chain-main/blob/master/CHANGELOG.md)?
- [ ] If you are contributing for the first time, please read the agreement in [CONTRIBUTING.md](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md) now and add a comment to this pull request stating that your PR is in accordance with the [Developer's Certificate of Origin](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md#developer-certificate-of-originn).

Thank you for your code, it's appreciated! :)

### CODE REVIEW

File: nix/default.nix [hunk 1]
```diff
@@ -1,8 +1,17 @@
...
+      (import (sources.dapptools + "/release.nix") { }).dapphub.linux.stable;
+    x86_64-darwin =
+      (import (sources.dapptools + "/release.nix") { }).dapphub.darwin.stable;
+  }.${system} or (throw
+    "Unsupported system: ${system}");
+in
 import sources.nixpkgs {
   overlays = [
-    (import (sources.dapptools + "/overlay.nix"))
+    (_: pkgs: dapptools)
```
Comment (yihuang) [inline nix/default.nix:14]:
```suggestion
    (_: pkgs: dapptools)  # use released version to hit the binary cache
```

**Review by yihuang — APPROVED** [review]
LGTM

**Comment by yihuang** [comment]
https://github.com/crypto-org-chain/cronos/runs/4283019200?check_suite_focus=true

Need to update the contract compiled artifacts.

Also add an entry in changelog, and mention the default solc version is changed.

**Review by leejw51crypto — APPROVED** [review]
lgtm

File: nix/default.nix (hunk 1 above)
Comment (damoncro) [reply nix/default.nix]: Done.

**Comment by damoncro** [comment]
Done. @yihuang

**Comment by yihuang** [comment]
There are some commits not signed. @damoncro

**Comment by damoncro** [comment]
All signed now. @yihuang

**Comment by thomas-nguy** [comment]
@damoncro can I merge your PR?

**Review by yihuang — APPROVED** [review]

**Comment by damoncro** [comment]
@yihuang @thomas-nguy Thanks

**Pull request merged**